operations/mimir-mixin-compiled/alerts.yaml

```
@@ -167,6 +167,29 @@ groups:
     for: 1h
     labels:
       severity: warning
+  - alert: MimirRulerInstanceHasNoRuleGroups
+    annotations:
+      message: Mimir ruler {{ $labels.pod }} in {{ $labels.cluster }}/{{ $labels.namespace
+        }} has no rule groups assigned.
+      runbook_url: https://grafana.com/docs/mimir/latest/operators-guide/mimir-runbooks/#mimirrulerinstancehasnorulegroups
+    expr: |
+      # Alert on ruler instances that have no rule groups assigned,
+      (min by(cluster, namespace, pod) (cortex_ruler_managers_total) == 0)
+      # but only if other ruler instances of the same cell do have rule groups assigned
+      and on (cluster, namespace)
+      (max by(cluster, namespace) (cortex_ruler_managers_total) > 0)
+      # but not if we have exactly 2 instances and the cell has less than 5 rule groups.
+      # We never want to scale to less than 2 instances because of availability concerns,
+      # and in these cases, it's likely that rule groups aren't sharded across both instances.
+      unless on (cluster, namespace)
+      (
+          count by (cluster, namespace) (cortex_ruler_managers_total) == 2
+          and on (cluster, namespace)
+          count by (cluster, namespace) (cortex_prometheus_rule_group_rules{job=~".+?((ruler|cortex|mimir|mimir-backend.*))"}) < 5
```

I would suggest to simplify this into:

```suggestion
          count by (cluster, namespace) (cortex_ruler_managers_total) <= 2
```

Rationale:
- We also don't want to alert if you run just 1 ruler replica
- I would skip the check on `cortex_prometheus_rule_group_rules`

---

Fixed this, but do we really want to run only one replica and accept the implications for availability of this component in case of instabilities of a node?

---

We recommend at least 2 ruler replicas, but this alert is unrelated from that and shouldn't fire if you only run 1.